src/pages/files.tsx

```diff
@@ -182,6 +182,38 @@ function FilesApp(props: IFilesAppProps) {
     void refreshListing();
   }, [refreshListing]);
 
+  // Show the native "Leave site?" prompt when there is at least one uploaded (supported) file.
+  const hasAnyFileBeenUploaded = React.useMemo(() => {
+    if (!listing || listing.type !== 'directory') {
+      return false;
+    }
+    const items = listing.content as Contents.IModel[];
+    return items.some(f => {
+      if (f.type !== 'file') {
+        return false;
+      }
+      return isSupportedFileType({
+        name: f.name,
+        type: f.mimetype ?? '',
+        size: f.size ?? 0,
+        lastModified: Date.now()
+      } as File);
+    });
+  }, [listing]);
+
+  useEffect(() => {
+    if (!hasAnyFileBeenUploaded) {
+      return;
+    }
+
+    const handler = (e: BeforeUnloadEvent) => {
+      e.preventDefault();
```

Comment:
Maybe just in case (some mobile browsers like Samsung only got support for `preventDefault()` way last year):

```suggestion
      e.preventDefault();
      return true;
```

Reply:
Thanks, I didn't know this! :)